@here/harp-mapview/lib/MapView.ts

```diff
@@ -2713,6 +2722,36 @@ export class MapView extends THREE.EventDispatcher {
             this.enqueueTextRendererCreation();
         }
 
+        // Check if this is the time to place the labels for the first time. Pretty much everything
+        // should have been loaded, and no animation should be running.
+        if (
+            !this.m_initialTextPlacementDone &&
+            !this.m_firstFrameComplete &&
+            !this.isDynamicFrame &&
+            !this.m_initialTextPlacementDone &&
+            !this.m_themeIsLoading &&
+            this.m_visibleTiles.allVisibleTilesLoaded &&
```

Comment:
IMO, ImageCache check is still missing from this "litany"

Reply:
Yes, a litany it is. ImageCache may be a bit tricky, because not every imageItem in it is being loaded, since that is optional. But the images for the POIs should be handled. I would rather add that check and eventually fix it once we have POIs to render in our default themes.